nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -531,24 +692,69 @@ mod test {
             .dns_config_read(&opctx, DnsGroup::External)
             .await
             .unwrap();
-        assert_eq!(dns_internal.generation, dns_external.generation);
+        // The external DNS zone has an extra update due to the initial Silo
+        // creation.
+        assert_eq!(dns_internal.generation + 1, dns_external.generation);
         assert_eq!(dns_internal.zones, dns_external.zones);
 
-        // It should also be idempotent.
-        let rack2 = datastore
-            .rack_set_initialized(
+        // Verify the details about the initial Silo.
+        let silos = datastore
+            .silos_list(
                 &opctx,
-                RackInit {
-                    rack_id: rack_id(),
-                    services,
-                    datasets,
-                    service_ip_pool_ranges,
-                    certificates,
-                    internal_dns: internal_dns_empty(),
-                    external_dns: external_dns_empty(),
+                &PaginatedBy::Name(DataPageParams {
+                    marker: None,
+                    limit: NonZeroU32::new(2).unwrap(),
+                    direction: dropshot::PaginationOrder::Ascending,
+                }),
+            )
+            .await
+            .expect("Failed to list Silos");
+        // It should *not* show up in the list because it's not discoverable.
+        assert_eq!(silos.len(), 0);
+        let (authz_silo, db_silo) = LookupPath::new(&opctx, &datastore)
+            .silo_name(&nexus_db_model::Name(
+                rack_init.recovery_silo.identity.name.clone(),
+            ))
+            .fetch()
+            .await
+            .expect("Failed to lookup Silo");
+        assert!(!db_silo.discoverable);
```

Comment:
Do we confirm that it's not possible for an operator to create a new silo with the same name as the recovery silo?

Reply:
Only by way of existing tests that verify that you cannot create a Silo with a name that conflicts with an existing Silo.  I could add an explicit test here.

Reply:
I filed #2985 for part of this.  I'm also adding a test that we cannot re-use the name while it's in use.  (Turns out there was a bug here where we would fail on a 500 because we were not handling the database error correctly.  This is not specific to the recovery Silo.)

Reply:
Test added in the latest commit.